File: 2292-counting-words-with-a-given-prefix/counting-words-with-a-given-prefix.cpp
class Solution {
public:
    bool prefix(string s,string t){
        if(s.size() < t.size()) return false;
        int i = 0,j=0;
        while(i<s.size() and j<t.size()){
            if(s[i]!=t[j]) return false;
            i++;
            j++;
        }
        return true;
    }
    int prefixCount(vector<string>& words, string pref) {
        int cnt = 0;
        for(auto it:words) if(prefix(it,pref)) cnt++;
        return cnt;
    }
};

File: 2292-counting-words-with-a-given-prefix/counting-words-with-a-given-prefix.rs
impl Solution {
    pub fn is_prefix(s:&str,t:&str) -> bool{
        if s.len() < t.len(){
            return false;
        }
        if s.starts_with(t){
            return true;
        }
        false
    }
    pub fn prefix_count(words: Vec<String>, pref: String) -> i32 {
        let mut count = 0;
        for i in 0..words.len(){
            if Self::is_prefix(&words[i],&pref){
                count += 1;
            }
        }
        count
    }
}